cli/internal/tablenamechanger/table_name_changer.go

```diff
@@ -0,0 +1,62 @@
+package tablenamechanger
+
+import (
+	"github.com/cloudquery/cloudquery/cli/internal/specs/v0"
+	"github.com/cloudquery/plugin-pb-go/pb/plugin/v3"
+	"github.com/cloudquery/plugin-sdk/v4/schema"
+)
+
+// TableNameChanger manages table name changes due to transformers, on a per-destination basis.
+// It can learn about table name changes and apply those changes on further steps.
+type TableNameChanger struct {
+	tableNameChanges map[string]map[string]string
```

Comment:
Should this be concurrent safe?

Reply:
No need. There are 3 places that update it. Two are on the same goroutine, and the third happens after a `Wait` on that one.

Reply:
But I can make it concurrency-safe 🤷 It's no big deal.

If you think about it, if we have to make it concurrency-safe, I think it's not safe, is it? 🤔 We don't want to update table names until they're all modified.

The per-destination multiplexing happens in loops, so it's all sequential.